src/CarbonAware.WebApi/src/Configuration/ServiceCollectionExtensions.cs

```diff
@@ -7,13 +9,25 @@ public static void AddMonitoringAndTelemetry(this IServiceCollection services, I
         
         var envVars = configuration?.GetSection(CarbonAwareVariablesConfiguration.Key).Get<CarbonAwareVariablesConfiguration>();
         var telemetryProvider = GetTelemetryProviderFromValue(envVars?.TelemetryProvider);
-
+        var logger = CreateConsoleLogger(configuration);
         switch (telemetryProvider) {
             case TelemetryProviderType.ApplicationInsights:
             {
-                if (!String.IsNullOrEmpty(configuration?["ApplicationInsights_Connection_String"]))
+                string? instrumentationKey = configuration?["AppInsights_InstrumentationKey"];
```

Comment:
> Created a variable since it was being used more than once in the code. Although it is only used twice, it is easier to change in one place and it less error prone.

It looks inconsistent. Why not to do the same with `configuration?["ApplicationInsights_Connection_String"]`